src/overlays/actors/ovl_Boss_Ganondrof/z_boss_ganondrof.c

```diff
@@ -61,71 +148,1395 @@ static ColliderCylinderInit D_80914CFC = {
+            if (this->timers[0] == 0) {
+                BossGanondrof_SetupCharge(this, globalCtx);
+            }
+            break;
+    }
+
+    Math_ApproachF(&thisx->world.pos.x, targetX, 0.05f, this->floatSpeed);
+    if (this->timers[2] != 0) {
+        Math_ApproachF(&thisx->world.pos.y, targetY + 100.0f, 0.1f, 50.0f);
+    } else {
+        Math_ApproachF(&thisx->world.pos.y, targetY, 0.05f, 10.0f);
+    }
+
+    Math_ApproachF(&thisx->world.pos.z, targetZ, 0.05f, this->floatSpeed);
+    Math_ApproachF(&this->floatSpeed, 50.0f, 1.0f, 0.5f);
+    thisx->velocity.x = thisx->world.pos.x - thisx->prevPos.x;
+    thisx->velocity.z = thisx->world.pos.z - thisx->prevPos.z;
+    thisx->world.pos.y += 2.0f * Math_SinS(this->varianceTimer * 1500);
+    Math_ApproachS(&thisx->shape.rot.y, thisx->yawTowardsPlayer, 5, 0xBB8);
+    if ((this->varianceTimer & 1) == 0) {
+        Vec3f sp5C;
+        Vec3f sp50 = { 0.0f, 0.0f, 0.0f };
+        Vec3f sp44 = { 0.0f, 0.0f, 0.0f };
+
+        for (i = 0; i < 3; i++) {
+            sp5C.x = Rand_CenteredFloat(20.0f) + this->spearTip.x;
+            sp5C.y = Rand_CenteredFloat(20.0f) + this->spearTip.y;
+            sp5C.z = Rand_CenteredFloat(20.0f) + this->spearTip.z;
+            sp44.y = -0.08f;
+            EffectSsFhgFlash_SpawnLightBall(globalCtx, &sp5C, &sp50, &sp44, (s16)(Rand_ZeroOne() * 80.0f) + 150,
+                                            FHGFLASH_LIGHTBALL_GREEN);
+        }
+    }
+
+    if (player->unk_A73 != 0) {
+        BossGanondrof_SetupBlock(this, globalCtx);
+    }
+
+    Audio_PlayActorSound2(thisx, NA_SE_EN_FANTOM_FLOAT - SFX_FLAG);
+}
+
+void BossGanondrof_SetupThrow(BossGanondrof* this, GlobalContext* globalCtx) {
+    EnfHG* horse;
+    s16 lightTime;
+
+    this->endFrame = Animation_GetLastFrame(&gPhantomGanonAnim_00EC94);
+    Animation_MorphToPlayOnce(&this->skelAnime, &gPhantomGanonAnim_00EC94, -5.0f);
+    this->actionFunc = BossGanondrof_Throw;
+    if ((Rand_ZeroOne() <= 0.1f) && (this->throwCount >= 10) && (this->flyMode == FHG_FLY_NEUTRAL)) {
+        this->actionState = THROW_SLOW;
+        this->throwFrame = 1000;
+        lightTime = 32;
+    } else {
+        this->actionState = THROW_NORMAL;
+        this->throwFrame = 25;
+        lightTime = 25;
+    }
+
+    horse = (EnfHG*)this->actor.child;
+    Actor_SpawnAsChild(&globalCtx->actorCtx, &this->actor, globalCtx, ACTOR_EN_FHG_FIRE, this->spearTip.x,
+                       this->spearTip.y, this->spearTip.z, lightTime, FHGFIRE_LIGHT_GREEN, 0, FHGFIRE_SPEAR_LIGHT);
+    this->actor.child = &horse->actor;
+    this->throwCount++;
+    Audio_PlayActorSound2(&this->actor, NA_SE_EN_FANTOM_STICK);
+}
+
+void BossGanondrof_Throw(BossGanondrof* this, GlobalContext* globalCtx) {
+    SkelAnime_Update(&this->skelAnime);
+    osSyncPrintf("this->fwork[GND_END_FRAME] = %d\n", (s16)this->endFrame);
+    osSyncPrintf("this->work[GND_SHOT_FRAME] = %d\n", this->throwFrame);
```

Comment:
we should discuss if we want to do the good ol `work` arrays lmfao
if it matches without its arguable that its easier to read. so probably. but yea idk for sure yet

Reply:
It will likely fix some awkward looking `(*this).whatever`, though `this->work[WHATEVER]` is also awkward.

Reply:
oh yea forgot it fixes that. the latter is def preferred then id say

Reply:
I'd have to change so much for that that I'd really prefer not to.

Reply:
`(*this).whatever` is a fake match to simulate an array access for ido. its def not what was wrote, so I think it really should be changed

Reply:
we can discuss using the `rName` macro system like from effects to improve readability. but yea i dont think we should keep the dereferences